Address more safer cpp warnings in WebKit/UIProcess/

cdumez opened the pull request:
#### 7fd0359ffa67a16628870bba5b09d84fd1db0a5c
<pre>
Address more safer cpp warnings in WebKit/UIProcess/
<a href="https://bugs.webkit.org/show_bug.cgi?id=301053">https://bugs.webkit.org/show_bug.cgi?id=301053</a>

Reviewed by Geoffrey Garen.

* Source/WebCore/Modules/notifications/NotificationData.h:
* Source/WebCore/Modules/notifications/NotificationDataCocoa.mm:
(WebCore::NotificationData::protectedDictionaryRepresentation const):
* Source/WebKit/SaferCPPExpectations/NoUnretainedMemberCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebKit/UIProcess/API/APIContextMenuElementInfoMac.h:
* Source/WebKit/UIProcess/API/APIFrameTreeNode.cpp:
(API::FrameTreeNode::protectedPage):
* Source/WebKit/UIProcess/API/APIFrameTreeNode.h:
* Source/WebKit/UIProcess/API/C/mac/WKNotificationPrivateMac.mm:
(WKNotificationCopyDictionaryRepresentation):
* Source/WebKit/UIProcess/API/C/mac/WKProtectionSpaceNS.mm:
(WKProtectionSpaceCopyNSURLProtectionSpace):
* Source/WebKit/UIProcess/API/Cocoa/WKSnapshotConfiguration.mm:
(-[WKSnapshotConfiguration dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/WKURLSchemeTask.mm:
(-[WKURLSchemeTaskImpl _requestOnlyIfCached]):
* Source/WebKit/UIProcess/API/Cocoa/WKUserScript.mm:
(-[WKUserScript initWithSource:injectionTime:forMainFrameOnly:inContentWorld:]):
(-[WKUserScript _userContentWorld]):
(-[WKUserScript _initWithSource:injectionTime:forMainFrameOnly:includeMatchPatternStrings:excludeMatchPatternStrings:associatedURL:contentWorld:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.mm:
(-[WKWebViewConfiguration _setAllowPostingLegacySynchronousMessages:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewTesting.mm:
(-[WKWebView _hasServiceWorkerBackgroundActivityForTesting]):
(-[WKWebView _hasServiceWorkerForegroundActivityForTesting]):
(-[WKWebView _propertiesOfLayerWithID:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataRecord.mm:
(-[WKWebsiteDataRecord dealloc]):
(-[WKWebsiteDataRecord description]):
* Source/WebKit/UIProcess/API/Cocoa/WKWindowFeatures.mm:
(-[WKWindowFeatures dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.mm:
(-[_WKApplicationManifest initWithCoder:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKArchiveConfiguration.mm:
(-[_WKArchiveConfiguration dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKAttachment.mm:
(-[_WKAttachmentInfo initWithAttachment:]):
(-[_WKAttachment dealloc]):
(-[_WKAttachment setFileWrapper:contentType:completion:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKAuthenticationExtensionsClientInputs.mm:
(-[_WKAuthenticationExtensionsClientInputs dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKAuthenticatorAssertionResponse.mm:
(-[_WKAuthenticatorAssertionResponse dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKAuthenticatorAttestationResponse.mm:
(-[_WKAuthenticatorAttestationResponse dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKAuthenticatorResponse.mm:
(-[_WKAuthenticatorResponse dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKAutomationSession.mm:
(protectedSession):
(-[_WKAutomationSession dealloc]):
(-[_WKAutomationSession setDelegate:]):
(-[_WKAutomationSession isPaired]):
(-[_WKAutomationSession isPendingTermination]):
(-[_WKAutomationSession isSimulatingUserInteraction]):
(-[_WKAutomationSession terminate]):
(-[_WKAutomationSession wasEventSynthesizedForAutomation:]):
(-[_WKAutomationSession markEventAsSynthesizedForAutomation:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKContentRuleListAction.mm:
(protectedAction):
(-[_WKContentRuleListAction dealloc]):
(-[_WKContentRuleListAction blockedLoad]):
(-[_WKContentRuleListAction blockedCookies]):
(-[_WKContentRuleListAction madeHTTPS]):
(-[_WKContentRuleListAction redirected]):
(-[_WKContentRuleListAction modifiedHeaders]):
(-[_WKContentRuleListAction notifications]):
* Source/WebKit/UIProcess/API/Cocoa/_WKContextMenuElementInfo.mm:
(-[_WKContextMenuElementInfo hitTestResult]):
* Source/WebKit/UIProcess/API/Cocoa/_WKCustomHeaderFields.mm:
(protectedFields):
(-[_WKCustomHeaderFields setFields:]):
(-[_WKCustomHeaderFields thirdPartyDomains]):
(-[_WKCustomHeaderFields setThirdPartyDomains:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKFeature.mm:
(-[_WKFeature dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKFrameTreeNode.mm:
(-[_WKFrameTreeNode dealloc]):
(-[_WKFrameTreeNode childFrames]):
* Source/WebKit/UIProcess/API/Cocoa/_WKJSHandle.mm:
(-[_WKJSHandle dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKProcessPoolConfiguration.mm:
(protectedProcessPoolConfiguration):
(-[_WKProcessPoolConfiguration dealloc]):
(-[_WKProcessPoolConfiguration setAdditionalReadAccessAllowedURLs:]):
(-[_WKProcessPoolConfiguration setCachePartitionedURLSchemes:]):
(-[_WKProcessPoolConfiguration setAlwaysRevalidatedURLSchemes:]):
(-[_WKProcessPoolConfiguration copyWithZone:]):
(-[_WKProcessPoolConfiguration setMemoryFootprintPollIntervalForTesting:]):
(-[_WKProcessPoolConfiguration setMemoryFootprintNotificationThresholds:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKRemoteWebInspectorViewController.mm:
(-[_WKRemoteWebInspectorViewController navigateExtensionTabWithIdentifier:toURL:completionHandler:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKRemoteWebInspectorViewControllerInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKResourceLoadStatisticsFirstParty.mm:
(-[_WKResourceLoadStatisticsFirstParty dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKResourceLoadStatisticsThirdParty.mm:
(-[_WKResourceLoadStatisticsThirdParty dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKSerializedNode.mm:
(-[_WKSerializedNode dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKTargetedElementInfo.mm:
(protectedInfo):
(-[_WKTargetedElementInfo boundsInWebView]):
(-[_WKTargetedElementInfo getChildFrames:]):
(-[_WKTargetedElementInfo isSameElement:]):
(-[_WKTargetedElementInfo takeSnapshotWithCompletionHandler:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKTargetedElementRequest.mm:
(protectedRequest):
(-[_WKTargetedElementRequest initWithSearchText:]):
(-[_WKTargetedElementRequest initWithPoint:]):
(-[_WKTargetedElementRequest initWithSelectors:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKUserStyleSheet.mm:
(-[_WKUserStyleSheet initWithSource:forWKWebView:forMainFrameOnly:includeMatchPatternStrings:excludeMatchPatternStrings:baseURL:level:contentWorld:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationAssertionResponse.mm:
(protectedResponse):
(-[_WKWebAuthenticationAssertionResponse dealloc]):
(-[_WKWebAuthenticationAssertionResponse userHandle]):
(-[_WKWebAuthenticationAssertionResponse credentialID]):
(-[_WKWebAuthenticationAssertionResponse setLAContext:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKWebPushMessage.mm:
(-[_WKWebPushMessage scopeURL]):
(-[_WKWebPushMessage scope]): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebsiteDataStoreConfiguration.mm:
(-[_WKWebsiteDataStoreConfiguration setProxyConfiguration:]):
* Source/WebKit/UIProcess/Automation/mac/WebAutomationSessionMac.mm:
(WebKit::WebAutomationSession::sendSynthesizedEventsToPage):
* Source/WebKit/UIProcess/Inspector/Cocoa/InspectorExtensionDelegate.mm:
(WebKit::InspectorExtensionDelegate::InspectorExtensionClient::didShowExtensionTab):
(WebKit::InspectorExtensionDelegate::InspectorExtensionClient::didHideExtensionTab):
(WebKit::InspectorExtensionDelegate::InspectorExtensionClient::didNavigateExtensionTab):
(WebKit::InspectorExtensionDelegate::InspectorExtensionClient::inspectedPageDidNavigate):
* Source/WebKit/UIProcess/Inspector/mac/WKInspectorViewController.mm:
(-[WKInspectorViewController webViewConfiguration]):
(-[WKInspectorViewController _horizontallyAttachedInspectedWebView]):
(-[WKInspectorViewController webViewWebContentProcessDidTerminate:]):
(-[WKInspectorViewController webView:decidePolicyForNavigationAction:decisionHandler:]):
(-[WKInspectorViewController inspectorWKWebViewDidBecomeActive:]):
(-[WKInspectorViewController inspectorWKWebView:willMoveToWindow:]):
(-[WKInspectorViewController inspectorWKWebViewDidMoveToWindow:]):
* Source/WebKit/UIProcess/WebURLSchemeTask.cpp:
(WebKit::WebURLSchemeTask::protectedNSRequest const):
* Source/WebKit/UIProcess/WebURLSchemeTask.h:

Canonical link: <a href="https://commits.webkit.org/301797@main">https://commits.webkit.org/301797@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/81d8fcf821120165f6d296ddd94a91d04100fbee

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/127088 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/46724 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/37826 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/134090 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/78650 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/26568b8a-9046-4b43-97aa-5b764ee89911) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/128959 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/47340 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/55250 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/96708 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/64752 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/341a6e02-2141-407d-8172-de3d94239a59) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/130036 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37896 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/113793 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/77218 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/3e24a013-25f9-4401-8071-fc0a11a85e40) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/36766 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/31925 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/77482 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/107763 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/32276 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/136616 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/53743 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/41395 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/105228 "58 failures") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/54247 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/110150 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/104917 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26756 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50446 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/28866 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/51286 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/53675 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/59548 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/52913 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/56246 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/54671 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren commented on Source/WebKit/UIProcess/API/Cocoa/_WKAttachment.mm:
```diff
@@ -123,7 +123,7 @@ - (void)dealloc
         return;
 
-    _attachment->~Attachment();
+    SUPPRESS_UNRETAINED_ARG _attachment->~Attachment();
```
I’m reading on a phone while feeding an infant so I might be misreading but:

If _attachment is a Ref, can’t we just rely on ObjC++ destruction to run its destructor? Running twice is technically undef. 

If not, maybe change to RefPtr and call clear() upon destruction?

This is not super important but it’s nice not to do surprising stuff in privileged code that we don’t test a lot.

cdumez replied:
It is not a Ref, I do think this is needed:
```
    AlignedStorage<API::Attachment> _attachment;
```

This is how most our API objects work.

geoffreygaren commented on Source/WebKit/UIProcess/API/Cocoa/_WKAutomationSession.mm:
```diff
@@ -62,8 +67,9 @@ - (void)dealloc
 
-    _session->setClient(nullptr);
-    _session->~WebAutomationSession();
+    Ref session = *_session;
```
Same comment about calling destructor.

geoffreygaren commented on Source/WebKit/UIProcess/API/Cocoa/_WKContentRuleListAction.mm:
```diff
@@ -29,22 +29,27 @@
         return;
 
-    _action->~ContentRuleListAction();
+    SUPPRESS_UNRETAINED_ARG _action->~ContentRuleListAction();
```
Same comment about calling destructor.

cdumez replied:
They are all stored in AlignedStorage.

geoffreygaren commented on Source/WebKit/UIProcess/API/Cocoa/_WKFeature.mm:
```diff
@@ -37,7 +37,7 @@ - (void)dealloc
         return;
 
-    _wrappedFeature->API::Feature::~Feature();
+    SUPPRESS_UNRETAINED_ARG _wrappedFeature->API::Feature::~Feature();
```
Same comment about calling destructor.

geoffreygaren commented on Source/WebKit/UIProcess/API/Cocoa/_WKFrameTreeNode.mm:
```diff
@@ -41,7 +41,7 @@ - (void)dealloc
     if (WebCoreObjCScheduleDeallocateOnMainRunLoop(_WKFrameTreeNode.class, self))
         return;
-    _node->API::FrameTreeNode::~FrameTreeNode();
+    SUPPRESS_UNRETAINED_ARG _node->API::FrameTreeNode::~FrameTreeNode();
```
Same comment about calling destructor.

geoffreygaren commented on Source/WebKit/UIProcess/API/Cocoa/_WKJSHandle.mm:
```diff
@@ -39,7 +39,7 @@ - (void)dealloc
     if (WebCoreObjCScheduleDeallocateOnMainRunLoop(_WKJSHandle.class, self))
         return;
-    _ref->API::JSHandle::~JSHandle();
+    SUPPRESS_UNRETAINED_ARG _ref->API::JSHandle::~JSHandle();
```
Calling destructor.

geoffreygaren commented on Source/WebKit/UIProcess/API/Cocoa/_WKProcessPoolConfiguration.mm:
```diff
@@ -51,7 +58,7 @@ - (void)dealloc
         return;
 
-    _processPoolConfiguration->~ProcessPoolConfiguration();
+    SUPPRESS_UNRETAINED_ARG _processPoolConfiguration->~ProcessPoolConfiguration();
```
Calling destructor.

geoffreygaren commented on Source/WebKit/UIProcess/API/Cocoa/_WKResourceLoadStatisticsFirstParty.mm:
```diff
@@ -34,7 +34,7 @@ - (void)dealloc
     if (WebCoreObjCScheduleDeallocateOnMainRunLoop(_WKResourceLoadStatisticsFirstParty.class, self))
         return;
-    _firstParty->API::ResourceLoadStatisticsFirstParty::~ResourceLoadStatisticsFirstParty();
+    SUPPRESS_UNRETAINED_ARG _firstParty->API::ResourceLoadStatisticsFirstParty::~ResourceLoadStatisticsFirstParty();
```
Calling destructor.

geoffreygaren commented on Source/WebKit/UIProcess/API/Cocoa/_WKResourceLoadStatisticsThirdParty.mm:
```diff
@@ -36,7 +36,7 @@ - (void)dealloc
     if (WebCoreObjCScheduleDeallocateOnMainRunLoop(_WKResourceLoadStatisticsThirdParty.class, self))
         return;
-    _thirdParty->API::ResourceLoadStatisticsThirdParty::~ResourceLoadStatisticsThirdParty();
+    SUPPRESS_UNRETAINED_ARG _thirdParty->API::ResourceLoadStatisticsThirdParty::~ResourceLoadStatisticsThirdParty();
```
Calling destructor.

geoffreygaren commented on Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationAssertionResponse.mm:
```diff
@@ -38,7 +47,7 @@ - (void)dealloc
         return;
 
-    _response->~WebAuthenticationAssertionResponse();
+    SUPPRESS_UNRETAINED_ARG _response->~WebAuthenticationAssertionResponse();
```
Calling destructor.

geoffreygaren commented on Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataRecord.mm:
```diff
@@ -67,7 +67,7 @@ - (void)dealloc
         return;
 
-    _websiteDataRecord->API::WebsiteDataRecord::~WebsiteDataRecord();
+    SUPPRESS_UNRETAINED_ARG _websiteDataRecord->API::WebsiteDataRecord::~WebsiteDataRecord();
```
Calling destructor.

geoffreygaren commented on Source/WebKit/UIProcess/API/Cocoa/WKWindowFeatures.mm:
```diff
@@ -39,7 +39,7 @@ - (void)dealloc
         return;
 
-    _windowFeatures->API::WindowFeatures::~WindowFeatures();
+    SUPPRESS_UNRETAINED_ARG _windowFeatures->API::WindowFeatures::~WindowFeatures();
```
Calling destructor.

geoffreygaren commented on Source/WebKit/UIProcess/Automation/mac/WebAutomationSessionMac.mm:
```diff
@@ -85,8 +85,8 @@
-    auto methodToSwizzle = class_getClassMethod(objc_getMetaClass(NSStringFromClass([NSEvent class]).UTF8String), @selector(pressedMouseButtons));
-    auto originalImplementation = method_setImplementation(methodToSwizzle, imp_implementationWithBlock([&mouseButtonsCurrentlyDown = m_mouseButtonsCurrentlyDown] {
+    auto methodToSwizzle = class_getClassMethod(RetainPtr { objc_getMetaClass(NSStringFromClass([NSEvent class]).UTF8String) }.get(), @selector(pressedMouseButtons));
+    SUPPRESS_UNRETAINED_ARG auto originalImplementation = method_setImplementation(methodToSwizzle, imp_implementationWithBlock([&mouseButtonsCurrentlyDown = m_mouseButtonsCurrentlyDown] {
```
I think this could use a comment about why suppress is ok.

cdumez replied:
This one is complaining about the Objective C block we're passing not being retained. The block gets constructed from a C++ lambda. Given that we're constructing the lambda/block on this line and it is only used on this line, I do not think we need to retain. It is also not clear to me how I would retain (using `BlockPtr` maybe?). To me this looks like a false positive.

geoffreygaren reviewed (approved):
r=me with some suggestions

geoffreygaren commented on Source/WebKit/UIProcess/API/Cocoa/_WKAttachment.mm:
```diff
@@ -57,7 +57,7 @@ - (instancetype)initWithAttachment:(const API::Attachment&)attachment
         return nil;
 
-    _attachment = attachment;
+    lazyInitialize(_attachment, Ref { attachment });
```
Since attachment is a function argument, and callers retain, why do we need to retain here? Is this a false positive?

cdumez replied:
This is not about static analysis. It doesn't build without it because of the way `lazyInitialize()` is written:
```
void lazyInitialize(const RefPtr<T>& ptr, Ref<U>&& obj);
```

Error:
```
In file included from /Volumes/Work/WebKit/OpenSource/WebKitBuild/Debug/DerivedSources/WebKit/unified-sources/UnifiedSource29-nonARC.mm:4:
/Volumes/Work/WebKit/OpenSource/Source/WebKit/UIProcess/API/Cocoa/_WKAttachment.mm:60:5: error: no matching function for call to 'lazyInitialize'
   60 |     lazyInitialize(_attachment, attachment);
/Volumes/Work/WebKit/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/RefPtr.h:271:20: note: candidate template ignored: could not match 'Ref<U>' against 'const API::Attachment'
  271 | ALWAYS_INLINE void lazyInitialize(const RefPtr<T>& ptr, Ref<U>&& obj)
```

cdumez committed:
Address more safer cpp warnings in WebKit/UIProcess/
https://bugs.webkit.org/show_bug.cgi?id=301053

Reviewed by Geoffrey Garen.

* Source/WebCore/Modules/notifications/NotificationData.h:
* Source/WebCore/Modules/notifications/NotificationDataCocoa.mm:
(WebCore::NotificationData::protectedDictionaryRepresentation const):
* Source/WebKit/SaferCPPExpectations/NoUnretainedMemberCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebKit/UIProcess/API/APIContextMenuElementInfoMac.h:
* Source/WebKit/UIProcess/API/APIFrameTreeNode.cpp:
(API::FrameTreeNode::protectedPage):
* Source/WebKit/UIProcess/API/APIFrameTreeNode.h:
* Source/WebKit/UIProcess/API/C/mac/WKNotificationPrivateMac.mm:
(WKNotificationCopyDictionaryRepresentation):
* Source/WebKit/UIProcess/API/C/mac/WKProtectionSpaceNS.mm:
(WKProtectionSpaceCopyNSURLProtectionSpace):
* Source/WebKit/UIProcess/API/Cocoa/WKSnapshotConfiguration.mm:
(-[WKSnapshotConfiguration dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/WKURLSchemeTask.mm:
(-[WKURLSchemeTaskImpl _requestOnlyIfCached]):
* Source/WebKit/UIProcess/API/Cocoa/WKUserScript.mm:
(-[WKUserScript initWithSource:injectionTime:forMainFrameOnly:inContentWorld:]):
(-[WKUserScript _userContentWorld]):
(-[WKUserScript _initWithSource:injectionTime:forMainFrameOnly:includeMatchPatternStrings:excludeMatchPatternStrings:associatedURL:contentWorld:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.mm:
(-[WKWebViewConfiguration _setAllowPostingLegacySynchronousMessages:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewTesting.mm:
(-[WKWebView _hasServiceWorkerBackgroundActivityForTesting]):
(-[WKWebView _hasServiceWorkerForegroundActivityForTesting]):
(-[WKWebView _propertiesOfLayerWithID:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataRecord.mm:
(-[WKWebsiteDataRecord dealloc]):
(-[WKWebsiteDataRecord description]):
* Source/WebKit/UIProcess/API/Cocoa/WKWindowFeatures.mm:
(-[WKWindowFeatures dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.mm:
(-[_WKApplicationManifest initWithCoder:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKArchiveConfiguration.mm:
(-[_WKArchiveConfiguration dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKAttachment.mm:
(-[_WKAttachmentInfo initWithAttachment:]):
(-[_WKAttachment dealloc]):
(-[_WKAttachment setFileWrapper:contentType:completion:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKAuthenticationExtensionsClientInputs.mm:
(-[_WKAuthenticationExtensionsClientInputs dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKAuthenticatorAssertionResponse.mm:
(-[_WKAuthenticatorAssertionResponse dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKAuthenticatorAttestationResponse.mm:
(-[_WKAuthenticatorAttestationResponse dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKAuthenticatorResponse.mm:
(-[_WKAuthenticatorResponse dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKAutomationSession.mm:
(protectedSession):
(-[_WKAutomationSession dealloc]):
(-[_WKAutomationSession setDelegate:]):
(-[_WKAutomationSession isPaired]):
(-[_WKAutomationSession isPendingTermination]):
(-[_WKAutomationSession isSimulatingUserInteraction]):
(-[_WKAutomationSession terminate]):
(-[_WKAutomationSession wasEventSynthesizedForAutomation:]):
(-[_WKAutomationSession markEventAsSynthesizedForAutomation:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKContentRuleListAction.mm:
(protectedAction):
(-[_WKContentRuleListAction dealloc]):
(-[_WKContentRuleListAction blockedLoad]):
(-[_WKContentRuleListAction blockedCookies]):
(-[_WKContentRuleListAction madeHTTPS]):
(-[_WKContentRuleListAction redirected]):
(-[_WKContentRuleListAction modifiedHeaders]):
(-[_WKContentRuleListAction notifications]):
* Source/WebKit/UIProcess/API/Cocoa/_WKContextMenuElementInfo.mm:
(-[_WKContextMenuElementInfo hitTestResult]):
* Source/WebKit/UIProcess/API/Cocoa/_WKCustomHeaderFields.mm:
(protectedFields):
(-[_WKCustomHeaderFields setFields:]):
(-[_WKCustomHeaderFields thirdPartyDomains]):
(-[_WKCustomHeaderFields setThirdPartyDomains:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKFeature.mm:
(-[_WKFeature dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKFrameTreeNode.mm:
(-[_WKFrameTreeNode dealloc]):
(-[_WKFrameTreeNode childFrames]):
* Source/WebKit/UIProcess/API/Cocoa/_WKJSHandle.mm:
(-[_WKJSHandle dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKProcessPoolConfiguration.mm:
(protectedProcessPoolConfiguration):
(-[_WKProcessPoolConfiguration dealloc]):
(-[_WKProcessPoolConfiguration setAdditionalReadAccessAllowedURLs:]):
(-[_WKProcessPoolConfiguration setCachePartitionedURLSchemes:]):
(-[_WKProcessPoolConfiguration setAlwaysRevalidatedURLSchemes:]):
(-[_WKProcessPoolConfiguration copyWithZone:]):
(-[_WKProcessPoolConfiguration setMemoryFootprintPollIntervalForTesting:]):
(-[_WKProcessPoolConfiguration setMemoryFootprintNotificationThresholds:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKRemoteWebInspectorViewController.mm:
(-[_WKRemoteWebInspectorViewController navigateExtensionTabWithIdentifier:toURL:completionHandler:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKRemoteWebInspectorViewControllerInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKResourceLoadStatisticsFirstParty.mm:
(-[_WKResourceLoadStatisticsFirstParty dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKResourceLoadStatisticsThirdParty.mm:
(-[_WKResourceLoadStatisticsThirdParty dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKSerializedNode.mm:
(-[_WKSerializedNode dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKTargetedElementInfo.mm:
(protectedInfo):
(-[_WKTargetedElementInfo boundsInWebView]):
(-[_WKTargetedElementInfo getChildFrames:]):
(-[_WKTargetedElementInfo isSameElement:]):
(-[_WKTargetedElementInfo takeSnapshotWithCompletionHandler:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKTargetedElementRequest.mm:
(protectedRequest):
(-[_WKTargetedElementRequest initWithSearchText:]):
(-[_WKTargetedElementRequest initWithPoint:]):
(-[_WKTargetedElementRequest initWithSelectors:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKUserStyleSheet.mm:
(-[_WKUserStyleSheet initWithSource:forWKWebView:forMainFrameOnly:includeMatchPatternStrings:excludeMatchPatternStrings:baseURL:level:contentWorld:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationAssertionResponse.mm:
(protectedResponse):
(-[_WKWebAuthenticationAssertionResponse dealloc]):
(-[_WKWebAuthenticationAssertionResponse userHandle]):
(-[_WKWebAuthenticationAssertionResponse credentialID]):
(-[_WKWebAuthenticationAssertionResponse setLAContext:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKWebPushMessage.mm:
(-[_WKWebPushMessage scopeURL]):
(-[_WKWebPushMessage scope]): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebsiteDataStoreConfiguration.mm:
(-[_WKWebsiteDataStoreConfiguration setProxyConfiguration:]):
* Source/WebKit/UIProcess/Automation/mac/WebAutomationSessionMac.mm:
(WebKit::WebAutomationSession::sendSynthesizedEventsToPage):
* Source/WebKit/UIProcess/Inspector/Cocoa/InspectorExtensionDelegate.mm:
(WebKit::InspectorExtensionDelegate::InspectorExtensionClient::didShowExtensionTab):
(WebKit::InspectorExtensionDelegate::InspectorExtensionClient::didHideExtensionTab):
(WebKit::InspectorExtensionDelegate::InspectorExtensionClient::didNavigateExtensionTab):
(WebKit::InspectorExtensionDelegate::InspectorExtensionClient::inspectedPageDidNavigate):
* Source/WebKit/UIProcess/Inspector/mac/WKInspectorViewController.mm:
(-[WKInspectorViewController webViewConfiguration]):
(-[WKInspectorViewController _horizontallyAttachedInspectedWebView]):
(-[WKInspectorViewController webViewWebContentProcessDidTerminate:]):
(-[WKInspectorViewController webView:decidePolicyForNavigationAction:decisionHandler:]):
(-[WKInspectorViewController inspectorWKWebViewDidBecomeActive:]):
(-[WKInspectorViewController inspectorWKWebView:willMoveToWindow:]):
(-[WKInspectorViewController inspectorWKWebViewDidMoveToWindow:]):
* Source/WebKit/UIProcess/WebURLSchemeTask.cpp:
(WebKit::WebURLSchemeTask::protectedNSRequest const):
* Source/WebKit/UIProcess/WebURLSchemeTask.h:

Canonical link: https://commits.webkit.org/301797@main
